indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -424,15 +424,16 @@ private void manageInternalCritical(
           catch (Exception e) {
             log.warn(e, "Exception thrown during isReady for task: %s", task.getId());
             final String errorMessage;
-            if (e instanceof MaxAllowedLocksExceededException) {
+            if (e instanceof MaxAllowedLocksExceededException || e instanceof DruidException) {
               errorMessage = e.getMessage();
             } else {
               errorMessage = StringUtils.format(
-                  "Encountered error[%s] while waiting for task to be ready. See Overlord logs for more details.",
-                  StringUtils.chop(e.getMessage(), 100)
+                  "Encountered error [%s] while waiting for task to be ready. See Overlord logs for more details.",
```

Comment:
Nit: Druid message formatting style does not use the space before the brackets.

```suggestion
                  "Encountered error[%s] while waiting for task to be ready. See Overlord logs for more details.",
```